lib/diff-array.js

```diff
@@ -191,6 +200,9 @@ var handleDeepProperties = function(object, paths, cb) {
     var currentObject = object;
 
     for (var i = 0; i < props.length - 1; i++) {
+      if(_.isUndefined(currentObject[props[i]])){
```

Comment:
Hmm if I remove the three lines in the handleDeepProperties for loop i get
```
TypeError: Cannot read property 'images' of undefined
    at http://localhost:3000/packages/urigo_angular.js?8ab7a0c439daa219c61958486f0f9d098d902f6a:573:36
    at Object.forEach (http://localhost:3000/packages/angular_angular.js?c4420bc04009b877e624e5f64e10fef52b19a1fe:380:20)
    at handleDeepProperties (http://localhost:3000/packages/urigo_angular.js?8ab7a0c439daa219c61958486f0f9d098d902f6a:565:11)
    at deepCopyObjectChanges (http://localhost:3000/packages/urigo_angular.js?8ab7a0c439daa219c61958486f0f9d098d902f6a:585:3)
    at Object.<anonymous> (http://localhost:3000/packages/local-test_urigo_angular.js?d7fce7d24c54fcf5502ee92c221fec2bb37a8d14:497:9)
    at attemptSync (http://localhost:3000/packages/sanjo_jasmine.js?ca8c9ae4782601ea2ea1c5a1b945c26fc83633ba:2286:24)
    at QueueRunner.run (http://localhost:3000/packages/sanjo_jasmine.js?ca8c9ae4782601ea2ea1c5a1b945c26fc83633ba:2274:9)
    at QueueRunner.execute (http://localhost:3000/packages/sanjo_jasmine.js?ca8c9ae4782601ea2ea1c5a1b945c26fc83633ba:2260:10)
    at Spec.Env.queueRunnerFactory (http://localhost:3000/packages/sanjo_jasmine.js?ca8c9ae4782601ea2ea1c5a1b945c26fc83633ba:1096:35)
    at Spec.execute (http://localhost:3000/packages/sanjo_jasmine.js?ca8c9ae4782601ea2ea1c5a1b945c26fc83633ba:845:10)
```

during the _should copy new values_ test